Exercise 1:
===
Write a short essay or report on the following topics:

What is data analysis?
Why is data analysis important in modern contexts?
List and describe three areas where data analysis is applied today.

________________________________________________________________________________________________

The Power of Insight: An Introduction to Modern Data Analysis

I. Defining the Digital Foundation


Data analysis is far more than just looking at numbers; it is the systematic process of collecting, cleaning, transforming, and interpreting data to discover useful information. In today’s world, this process is essential for identifying patterns and supporting complex decision-making. By transforming raw, unstructured data into actionable insights, organizations can solve deep-seated problems, evaluate their current performance, and make accurate predictions about future outcomes.

II. The Strategic Importance in a Modern Context

In the current technological landscape, data analytics is revolutionizing industries by empowering organizations to move away from "gut feelings" and toward smarter, data-driven decisions. This transformation is driving measurable gains in productivity and innovation. For instance, businesses adopting these approaches saw a 63% boost in productivity recently. With the global data analytics market projected to hit $132.9 billion by 2026, it is clear that analytics is a transformative force rather than a passing trend.


Beyond mere profit, data analysis is vital because it addresses four key pillars of modern strategy:

Informed Decision-Making: Using facts instead of assumptions.

Business Intelligence: Understanding customer preferences to stay ahead of competitors.

Performance Evaluation: Monitoring how well products or strategies are actually working.

Risk Management: Identifying potential challenges before they escalate into crises.

III. Applications Across Diverse Sectors

The practical application of these concepts varies by industry, but the core goal remains the same: clarity and efficiency.

In Business and Retail, raw data is the "new oil." By maintaining detailed records of potential customers and their behaviors, companies use analytics tools to unlock the potential of raw information, converting it into personalized marketing and better inventory management.

In Healthcare, data has become life-saving. Modern medical devices are now big-data oriented, allowing doctors to monitor critical patient information—like heart rate and temperature—through wearable technology. These records allow for more precise treatments and better patient outcomes.

Similarly, in Media and Entertainment, the business model now relies on analyzing content consumption. By gathering minute details on audience preferences, platforms can forecast what a target audience wants, optimizing content distribution and suggesting shows on-demand to increase viewer retention.

IV. Future Horizons: Trends for 2026 and Beyond

As we look toward the future, data analysis is shifting from retrospective reporting to real-time, autonomous systems. We are entering an era of Agentic AI, where systems can independently plan and execute analyses without waiting for human input. Furthermore, Data Democratization is making these powerful tools accessible to non-technical users through no-code platforms, allowing everyone in an organization to "ask" questions of their data.

V. Conclusion

Data analysis has evolved from a back-office function into the very heartbeat of modern society. Whether it is improving a business process, saving a life in a hospital, or predicting the next hit show, the ability to interpret data is the most critical skill of the digital age. As we move toward 2026, the integration of AI and real-time streaming will only continue to redefine how we understand and interact with the world around us.

________________________________________________

Exercise 2: Dataset Loading and Initial Analysis 
===

For the following dataset : How Much Sleep Do Americans Really Get?, Global Trends in Mental Health Disorder and Credit Card Approvals.

- Load the dataset 
- Display the first few rows.
- Provide a brief dataset description.

In [2]:
import pandas as pd

Loading the datasets:

In [3]:
df_sleep = pd.read_csv('sleep.csv')
df_mental = pd.read_csv('mental_health.csv')
df_credit = pd.read_csv('credit.csv')

print("Datasets loaded successfully!")

Datasets loaded successfully!


/var/folders/11/01wml45d32l3bgdl2z4gyjth0000gn/T/ipykernel_43317/1063486514.py:2: DtypeWarning: Columns (0: Bipolar disorder (%), 1: Eating disorders (%)) have mixed types. Specify dtype option on import or set low_memory=False.
  df_mental = pd.read_csv('mental_health.csv')


First 5 rows of Sleep dataset:

In [4]:
df_sleep.head()


,Person ID,Gender,Age,Occupation,Sleep Duration,Quality of Sleep,Physical Activity Level,Stress Level,BMI Category,Blood Pressure,Heart Rate,Daily Steps,Sleep Disorder
0,1,Male,27,Software Engineer,6.1,6,42,6,Overweight,126/83,77,4200,NaN
1,2,Male,28,Doctor,6.2,6,60,8,Normal,125/80,75,10000,NaN
2,3,Male,28,Doctor,6.2,6,60,8,Normal,125/80,75,10000,NaN
3,4,Male,28,Sales Representative,5.9,4,30,8,Obese,140/90,85,3000,Sleep Apnea
4,5,Male,28,Sales Representative,5.9,4,30,8,Obese,140/90,85,3000,Sleep Apnea


Description of Sleep dataset:

In [5]:
df_sleep.info()

<class 'pandas.DataFrame'>
RangeIndex: 374 entries, 0 to 373
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Person ID                374 non-null    int64  
 1   Gender                   374 non-null    str    
 2   Age                      374 non-null    int64  
 3   Occupation               374 non-null    str    
 4   Sleep Duration           374 non-null    float64
 5   Quality of Sleep         374 non-null    int64  
 6   Physical Activity Level  374 non-null    int64  
 7   Stress Level             374 non-null    int64  
 8   BMI Category             374 non-null    str    
 9   Blood Pressure           374 non-null    str    
 10  Heart Rate               374 non-null    int64  
 11  Daily Steps              374 non-null    int64  
 12  Sleep Disorder           155 non-null    str    
dtypes: float64(1), int64(7), str(5)
memory usage: 38.1 KB


First 5 rows of Mental Health dataset:

In [6]:
df_mental.head()

,index,Entity,Code,Year,Schizophrenia (%),Bipolar disorder (%),Eating disorders (%),Anxiety disorders (%),Drug use disorders (%),Depression (%),Alcohol use disorders (%)
0,0,Afghanistan,AFG,1990,0.16056,0.697779,0.101855,4.828830,1.677082,4.071831,0.672404
1,1,Afghanistan,AFG,1991,0.160312,0.697961,0.099313,4.829740,1.684746,4.079531,0.671768
2,2,Afghanistan,AFG,1992,0.160135,0.698107,0.096692,4.831108,1.694334,4.088358,0.670644
3,3,Afghanistan,AFG,1993,0.160037,0.698257,0.094336,4.830864,1.705320,4.096190,0.669738
4,4,Afghanistan,AFG,1994,0.160022,0.698469,0.092439,4.829423,1.716069,4.099582,0.669260


Description of Mental Health dataset:

In [7]:
df_mental.info()


<class 'pandas.DataFrame'>
RangeIndex: 108553 entries, 0 to 108552
Data columns (total 11 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   index                      108553 non-null  int64  
 1   Entity                     108553 non-null  str    
 2   Code                       103141 non-null  str    
 3   Year                       108553 non-null  str    
 4   Schizophrenia (%)          25875 non-null   str    
 5   Bipolar disorder (%)       19406 non-null   object 
 6   Eating disorders (%)       100236 non-null  object 
 7   Anxiety disorders (%)      6468 non-null    float64
 8   Drug use disorders (%)     6468 non-null    float64
 9   Depression (%)             6468 non-null    float64
 10  Alcohol use disorders (%)  6468 non-null    float64
dtypes: float64(4), int64(1), object(2), str(4)
memory usage: 9.1+ MB


First 5 rows of Credit dataset:

In [8]:
df_credit.head()

,Gender,Age,Debt,Married,BankCustomer,Industry,Ethnicity,YearsEmployed,PriorDefault,Employed,CreditScore,DriversLicense,Citizen,ZipCode,Income,Approved
0,1,30.83,0.000,1,1,Industrials,White,1.25,1,1,1,0,ByBirth,202,0,1
1,0,58.67,4.460,1,1,Materials,Black,3.04,1,1,6,0,ByBirth,43,560,1
2,0,24.50,0.500,1,1,Materials,Black,1.50,1,0,0,0,ByBirth,280,824,1
3,1,27.83,1.540,1,1,Industrials,White,3.75,1,1,5,1,ByBirth,100,3,1
4,1,20.17,5.625,1,1,Industrials,White,1.71,1,0,0,0,ByOtherMeans,120,0,1


Description of Credit dataset:

In [9]:
df_credit.info()

<class 'pandas.DataFrame'>
RangeIndex: 690 entries, 0 to 689
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Gender          690 non-null    int64  
 1   Age             690 non-null    float64
 2   Debt            690 non-null    float64
 3   Married         690 non-null    int64  
 4   BankCustomer    690 non-null    int64  
 5   Industry        690 non-null    str    
 6   Ethnicity       690 non-null    str    
 7   YearsEmployed   690 non-null    float64
 8   PriorDefault    690 non-null    int64  
 9   Employed        690 non-null    int64  
 10  CreditScore     690 non-null    int64  
 11  DriversLicense  690 non-null    int64  
 12  Citizen         690 non-null    str    
 13  ZipCode         690 non-null    int64  
 14  Income          690 non-null    int64  
 15  Approved        690 non-null    int64  
dtypes: float64(3), int64(10), str(3)
memory usage: 86.4 KB


______________________________

Exercise 3: Identifyng Data Types
===

For the datasets from the previous exercise, categorize each column of it as either quantitative or qualitative and explain your reasoning.

In [1]:
import pandas as pd

df_sleep = pd.read_csv('sleep.csv')
df_mental = pd.read_csv('mental_health.csv')
df_credit = pd.read_csv('credit.csv')

print("Datasets loaded successfully from the Ex_2 folder!")

Datasets loaded successfully from the Ex_2 folder!


/var/folders/11/01wml45d32l3bgdl2z4gyjth0000gn/T/ipykernel_43727/3308749529.py:4: DtypeWarning: Columns (0: Bipolar disorder (%), 1: Eating disorders (%)) have mixed types. Specify dtype option on import or set low_memory=False.
  df_mental = pd.read_csv('mental_health.csv')


In [2]:
df_sleep.info()

<class 'pandas.DataFrame'>
RangeIndex: 374 entries, 0 to 373
Data columns (total 13 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Person ID                374 non-null    int64  
 1   Gender                   374 non-null    str    
 2   Age                      374 non-null    int64  
 3   Occupation               374 non-null    str    
 4   Sleep Duration           374 non-null    float64
 5   Quality of Sleep         374 non-null    int64  
 6   Physical Activity Level  374 non-null    int64  
 7   Stress Level             374 non-null    int64  
 8   BMI Category             374 non-null    str    
 9   Blood Pressure           374 non-null    str    
 10  Heart Rate               374 non-null    int64  
 11  Daily Steps              374 non-null    int64  
 12  Sleep Disorder           155 non-null    str    
dtypes: float64(1), int64(7), str(5)
memory usage: 38.1 KB


In [3]:
print("--- Sleep Dataset Types ---")
print(df_sleep.dtypes)

--- Sleep Dataset Types ---
Person ID                    int64
Gender                         str
Age                          int64
Occupation                     str
Sleep Duration             float64
Quality of Sleep             int64
Physical Activity Level      int64
Stress Level                 int64
BMI Category                   str
Blood Pressure                 str
Heart Rate                   int64
Daily Steps                  int64
Sleep Disorder                 str
dtype: object


In [4]:
quant_sleep = df_sleep.select_dtypes(include=['number']).columns.tolist()
qual_sleep = df_sleep.select_dtypes(exclude=['number']).columns.tolist()

print(f"Quantitative = {quant_sleep}")
print(f"\nQualitative = {qual_sleep}")

Quantitative = ['Person ID', 'Age', 'Sleep Duration', 'Quality of Sleep', 'Physical Activity Level', 'Stress Level', 'Heart Rate', 'Daily Steps']

Qualitative = ['Gender', 'Occupation', 'BMI Category', 'Blood Pressure', 'Sleep Disorder']


In [5]:
df_mental.info()

<class 'pandas.DataFrame'>
RangeIndex: 108553 entries, 0 to 108552
Data columns (total 11 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   index                      108553 non-null  int64  
 1   Entity                     108553 non-null  str    
 2   Code                       103141 non-null  str    
 3   Year                       108553 non-null  str    
 4   Schizophrenia (%)          25875 non-null   str    
 5   Bipolar disorder (%)       19406 non-null   object 
 6   Eating disorders (%)       100236 non-null  object 
 7   Anxiety disorders (%)      6468 non-null    float64
 8   Drug use disorders (%)     6468 non-null    float64
 9   Depression (%)             6468 non-null    float64
 10  Alcohol use disorders (%)  6468 non-null    float64
dtypes: float64(4), int64(1), object(2), str(4)
memory usage: 9.1+ MB


In [6]:
df_mental.head()

,index,Entity,Code,Year,Schizophrenia (%),Bipolar disorder (%),Eating disorders (%),Anxiety disorders (%),Drug use disorders (%),Depression (%),Alcohol use disorders (%)
0,0,Afghanistan,AFG,1990,0.16056,0.697779,0.101855,4.828830,1.677082,4.071831,0.672404
1,1,Afghanistan,AFG,1991,0.160312,0.697961,0.099313,4.829740,1.684746,4.079531,0.671768
2,2,Afghanistan,AFG,1992,0.160135,0.698107,0.096692,4.831108,1.694334,4.088358,0.670644
3,3,Afghanistan,AFG,1993,0.160037,0.698257,0.094336,4.830864,1.705320,4.096190,0.669738
4,4,Afghanistan,AFG,1994,0.160022,0.698469,0.092439,4.829423,1.716069,4.099582,0.669260


In [7]:
print("--- Mental Health Data Types ---")
print(df_mental.dtypes)

--- Mental Health Data Types ---
index                          int64
Entity                           str
Code                             str
Year                             str
Schizophrenia (%)                str
Bipolar disorder (%)          object
Eating disorders (%)          object
Anxiety disorders (%)        float64
Drug use disorders (%)       float64
Depression (%)               float64
Alcohol use disorders (%)    float64
dtype: object


In [8]:
df_mental['index'] = df_mental['index'].astype(str)

quant_mental = df_mental.select_dtypes(include=['number']).columns.tolist()

qual_mental = df_mental.select_dtypes(exclude=['number']).columns.tolist()

print(f"Quantitative: {quant_mental}")
print(f"\nQualitative: {qual_mental}")

Quantitative: ['Anxiety disorders (%)', 'Drug use disorders (%)', 'Depression (%)', 'Alcohol use disorders (%)']

Qualitative: ['index', 'Entity', 'Code', 'Year', 'Schizophrenia (%)', 'Bipolar disorder (%)', 'Eating disorders (%)']


In [9]:
df_credit.info()

<class 'pandas.DataFrame'>
RangeIndex: 690 entries, 0 to 689
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Gender          690 non-null    int64  
 1   Age             690 non-null    float64
 2   Debt            690 non-null    float64
 3   Married         690 non-null    int64  
 4   BankCustomer    690 non-null    int64  
 5   Industry        690 non-null    str    
 6   Ethnicity       690 non-null    str    
 7   YearsEmployed   690 non-null    float64
 8   PriorDefault    690 non-null    int64  
 9   Employed        690 non-null    int64  
 10  CreditScore     690 non-null    int64  
 11  DriversLicense  690 non-null    int64  
 12  Citizen         690 non-null    str    
 13  ZipCode         690 non-null    int64  
 14  Income          690 non-null    int64  
 15  Approved        690 non-null    int64  
dtypes: float64(3), int64(10), str(3)
memory usage: 86.4 KB


In [10]:
df_credit.head()

,Gender,Age,Debt,Married,BankCustomer,Industry,Ethnicity,YearsEmployed,PriorDefault,Employed,CreditScore,DriversLicense,Citizen,ZipCode,Income,Approved
0,1,30.83,0.000,1,1,Industrials,White,1.25,1,1,1,0,ByBirth,202,0,1
1,0,58.67,4.460,1,1,Materials,Black,3.04,1,1,6,0,ByBirth,43,560,1
2,0,24.50,0.500,1,1,Materials,Black,1.50,1,0,0,0,ByBirth,280,824,1
3,1,27.83,1.540,1,1,Industrials,White,3.75,1,1,5,1,ByBirth,100,3,1
4,1,20.17,5.625,1,1,Industrials,White,1.71,1,0,0,0,ByOtherMeans,120,0,1


In [11]:
cols_to_fix = ['ZipCode', 'PriorDefault', 'Employed'] 

for col in cols_to_fix:
    if col in df_credit.columns:
        df_credit[col] = df_credit[col].astype(str)

quant_credit = df_credit.select_dtypes(include=['number']).columns.tolist()
qual_credit = df_credit.select_dtypes(exclude=['number']).columns.tolist()

print(f"Quantitative: {quant_credit}")
print(f"Qualitative: {qual_credit}")

Quantitative: ['Gender', 'Age', 'Debt', 'Married', 'BankCustomer', 'YearsEmployed', 'CreditScore', 'DriversLicense', 'Income', 'Approved']
Qualitative: ['Industry', 'Ethnicity', 'PriorDefault', 'Employed', 'Citizen', 'ZipCode']


Reasons:

Quantitative: These columns represent measurable magnitudes or counts, such as Age, Sleep Duration, and Income. They are categorized as quantitative because their numerical values exist on a scale where mathematical operations—like calculating an average or a sum provide meaningful insights into the data's physical or financial properties.

Qualitative: These columns represent descriptive labels or characteristics, such as Gender, Country, Zip Code, and Binary (0/1) flags. They are categorized as qualitative because they function as nominal identifiers to group observations. Even when they contain digits, performing arithmetic on them (like adding two Zip Codes) is mathematically meaningless because they identify "what kind" rather than "how much.

__________________________________________________

Exercise 4: Exploring Data Types
===

- Load the Iris dataset using Kaggle into a Jupyter Notebook
- Identify and list which columns in your dataset are qualitative and which are quantitative.
- Write a brief description of why each column is classified as qualitative or quantitative.

In [12]:
import pandas as pd

In [13]:
df_iris = pd.read_csv('iris.csv')

print("Dataset loaded successfully!")

Dataset loaded successfully!


First rows of Iris dataset:

In [14]:
df_iris.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,1,5.1,3.5,1.4,0.2,Iris-setosa
1,2,4.9,3.0,1.4,0.2,Iris-setosa
2,3,4.7,3.2,1.3,0.2,Iris-setosa
3,4,4.6,3.1,1.5,0.2,Iris-setosa
4,5,5.0,3.6,1.4,0.2,Iris-setosa


Quantitative and Qualitative columns:

In [15]:
df_iris['Id'] = df_iris['Id'].astype(str)

quant_iris = df_iris.select_dtypes(include=['number']).columns.tolist()
qual_iris = df_iris.select_dtypes(exclude=['number']).columns.tolist()

print(f"Quantitative {quant_iris}")
print(f"\nQualitative: {qual_iris}")

Quantitative ['SepalLengthCm', 'SepalWidthCm', 'PetalLengthCm', 'PetalWidthCm']

Qualitative: ['Id', 'Species']


Quantitative:
The columns SepalLengthCm, SepalWidthCm, PetalLengthCm, and PetalWidthCm are categorized as Quantitative because they represent physical measurements. Calculating an average for these columns is mathematically meaningful because it helps describe the "typical" size of a flower in the dataset.

Qualitative:
The columns Species and Id are categorized as Qualitative because they represent nominal identifiers.

- Species labels the group a flower belongs to (e.g., Iris-setosa).

- Id is a unique numeric label used for identification rather than measurement.

These columns are qualitative because performing arithmetic on them—such as adding two IDs together or averaging the species names—would result in values that have no real-world meaning.

_______________________________________

Exercise 5: Basic Observation Skills in Data Analysis
===

Task:

- Load the How Much Sleep Do Americans Really Get? dataset into Jupyter.

- Identify columns that could be interesting for a specific type of analysis (e.g., trend analysis, group comparison) and explain your choice.

In [16]:
import pandas as pd

In [17]:
df_american_sleep = pd.read_csv('american_sleep.csv')

print("Dataset loaded successfully!")

Dataset loaded successfully!


In [18]:
df_american_sleep.info()

<class 'pandas.DataFrame'>
RangeIndex: 945 entries, 0 to 944
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   index                     945 non-null    int64  
 1   Year                      945 non-null    int64  
 2   Period                    945 non-null    str    
 3   Avg hrs per day sleeping  945 non-null    float64
 4   Standard Error            945 non-null    float64
 5   Type of Days              945 non-null    str    
 6   Age Group                 945 non-null    str    
 7   Activity                  945 non-null    str    
 8   Sex                       945 non-null    str    
dtypes: float64(2), int64(2), str(5)
memory usage: 66.6 KB


In [19]:
df_american_sleep.head()

,index,Year,Period,Avg hrs per day sleeping,Standard Error,Type of Days,Age Group,Activity,Sex
0,0,2003,Annual,8.57,0.018,All days,15 years and over,Sleeping,Both
1,1,2004,Annual,8.55,0.026,All days,15 years and over,Sleeping,Both
2,2,2005,Annual,8.62,0.023,All days,15 years and over,Sleeping,Both
3,3,2006,Annual,8.63,0.024,All days,15 years and over,Sleeping,Both
4,4,2007,Annual,8.57,0.024,All days,15 years and over,Sleeping,Both


Analysis:

In [20]:
df_american_sleep.describe()

,index,Year,Avg hrs per day sleeping,Standard Error
count,945.000000,945.000000,945.000000,945.000000
mean,472.000000,2010.000000,8.806868,0.086958
std,272.942302,4.322782,0.545001,0.034631
min,0.000000,2003.000000,7.720000,0.018000
25%,236.000000,2006.000000,8.360000,0.063000
50%,472.000000,2010.000000,8.810000,0.084000
75%,708.000000,2014.000000,9.190000,0.106000
max,944.000000,2017.000000,10.650000,0.228000


Mean: The average American in this dataset sleeps about 8.80 hours.

Min/Max: Some people sleep 7.72 hours, while others sleep up to 10.65 hours.

Year: The data covers a range from 2003 to 2017.

In [21]:
df_american_sleep['Sex'].unique()

<StringArray>
['Both', 'Men', 'Women']
Length: 3, dtype: str

In [22]:
df_american_sleep['Age Group'].unique()

<StringArray>
['15 years and over',    '15 to 24 years',    '25 to 34 years',
    '35 to 44 years',    '45 to 54 years',    '55 to 64 years',
 '65 years and over']
Length: 7, dtype: str

Group Comparison and Trend Analysis:

Columns: Sex, Age Group, and Avg hrs per day sleeping.

I will focus specifically on the 'Men' and 'Women' categories to observe the differences in their sleep patterns

Explanation: I chose Sex because ".unique()" showed that the data is divided into 'Men' and 'Women,' which is perfect for a Group Comparison to see who sleeps more.

I chose Age Group because it allows for a Trend Analysis to see if sleep changes as people move from '15-24 years' to '65 years and over.'

I chose Avg hrs per day sleeping because my .describe() results showed a wide range of sleep times (from 7.7 to 10.6 hours), and I want to see how that variety is distributed among the different groups.

In [23]:
women_sleep = df_american_sleep[df_american_sleep['Sex'] == 'Women']
print("Average sleep for Women:", women_sleep['Avg hrs per day sleeping'].mean())

men_sleep = df_american_sleep[df_american_sleep['Sex'] == 'Men']
print("\nAverage sleep for Men:", men_sleep['Avg hrs per day sleeping'].mean())

Average sleep for Women: 8.881174603174602

Average sleep for Men: 8.7315873015873


_________________________________________

Exercise 6: Identifying Data Types
===

Below are various data sources. Identify whether each one is an example of structured or unstructured data.

- A company’s financial reports stored in an Excel file.
- Photographs uploaded to a social media platform.
- A collection of news articles on a website.
- Inventory data in a relational database.
- Recorded interviews from a market research study.

A company’s financial reports in an Excel file:
________________________________________________
 Classification: Structured Data 

 Financial reports in Excel are organized into rows and columns with clear headers (like "Revenue" or "Date"). Because each piece of data has a fixed place and a specific format (usually numbers or dates), a computer can easily search and calculate it.
 _______________________________________________

Photographs uploaded to a social media platform:
___________________________________________________
Classification: Unstructured Data 

Photos do not have a pre-defined row and column structure. While they contain a lot of information (colors, faces, locations), the data is "random" to a computer until it uses special AI to "read" what is inside the image.
_________________________________________________

A collection of news articles on a website:
_______________________________________________
Classification: Unstructured Data 

Even though an article has a title and a body, the actual content is "free-form" text. Because the length, vocabulary, and sentence structure change every time, it doesn't fit into a tidy spreadsheet.
________________________________________________

Inventory data in a relational database:
__________________________________________
Classification: Structured Data 

 Relational databases (like SQL) are the definition of structured data. They use strict tables where every "Product ID" or "Quantity" must follow a specific rule, making it very easy to manage and analyze.
 ___________________________________________

Recorded interviews from a market research study:

Classification: Unstructured Data 

Audio files are similar to photos. The data consists of sounds, tones, and pauses. There is no internal "table" for the information, so it requires transcription or specialized software to turn it into something a computer can analyze as "data."

___________________________________________________

Exercise 7: Transformation Exercise
====
_______________________________________
For each of the following unstructured data sources, propose a method to convert it into structured data. Explain your reasoning.

- A series of blog posts about travel experiences.
- Audio recordings of customer service calls.
- Handwritten notes from a brainstorming session.
- A video tutorial on cooking.


Blog posts about travel experiences:
____________________________________________
Transformation Method: Use Sentiment Analysis and Keyword Extraction.

I can create a spreadsheet with columns like Destination, Rating (1-5), and Sentiment (Positive/Negative). By scanning the text for specific city names and "mood" words (like "amazing" vs "crowded"), you turn a long story into a row of data that you can filter and count.
____________________________________

Audio recordings of customer service calls:
__________________________________________
Transformation Method: Speech-to-Text Transcription followed by Categorization.

First, convert the audio into text. Then, create a table with columns like Call Duration, Reason for Call (e.g., Refund, Technical, Billing), and Resolution (Yes/No). This turns a sound file into a list of searchable issues that a manager can analyze.
____________________________________________

Handwritten notes from a brainstorming session:
_____________________________________________
Transformation Method: OCR (Optical Character Recognition) and Thematic Coding.

Use a scanner or app to turn the handwriting into digital text. Then, you can categorize the ideas into a table with columns like Idea Owner, Category (e.g., Marketing, Product, Design), and Priority Level. This makes a messy page of notes easy to sort in Excel.
______________________________________________

A video tutorial on cooking:
_________________________________
Transformation Method: Object Detection and Step-by-Step Logging.

You can watch the video and extract the data into a spreadsheet with columns like Ingredient Name, Quantity, Cooking Technique (e.g., Frying, Boiling), and Time Spent. This turns a 10-minute visual experience into a structured recipe or a data list of movements and tools used.

______________________________________

Exercise 8 : Import a file from Kaggle
====
- Please download the dataset directly from the GitHub repository to use it in this project.

- Import the train dataset. Use the train.csv file.

- Print the first few rows of the DataFrame.

In [24]:
import pandas as pd

In [25]:
df_train = pd.read_csv('train.csv')

print("Dataset loaded successfully!")

Dataset loaded successfully!


In [26]:
df_train.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


______________________________________

Exercise 9 : Export a dataframe to excel format 
and JSON format
====
- Create a simple dataframe.

- Export the dataframe to an excel file.

- Export the dataframe to a JSON file.

In [27]:
%pip install openpyxl

Note: you may need to restart the kernel to use updated packages.


In [28]:
import pandas as pd

In [29]:
data = {
    'Name': ['Alme', 'Itay', 'Lancelot'],
    'Score': [95, 88, 100]
}

df_mydata = pd.DataFrame(data)

Creating an Excel:

In [30]:
df_mydata.to_excel('my_exercise_data.xlsx', index=False)

print("Excel file created!")

Excel file created!


Creating a Json file:

In [31]:
df_mydata.to_json('my_exercise_data.json')

print("JSON file created!")

JSON file created!


In [33]:
print("Success!")

Success!


_________________________________

Exercise 10: Reading JSON Data
===
Use a sample JSON dataset

- Import the JSON data from the provided URL.

- Use Pandas to read the JSON data.

- Display the first five entries of the data.

In [34]:
import pandas as pd

In [35]:
df_posts = pd.read_json('posts.json')

print("Dataset loaded successfully!")

Dataset loaded successfully!


In [36]:
df_posts.head()

,userId,id,title,body
0,1,1,sunt aut facere repellat provident occaecati e...,quia et suscipit\nsuscipit recusandae consequu...
1,1,2,qui est esse,est rerum tempore vitae\nsequi sint nihil repr...
2,1,3,ea molestias quasi exercitationem repellat qui...,et iusto sed quo iure\nvoluptatem occaecati om...
3,1,4,eum et est occaecati,ullam et saepe reiciendis voluptatem adipisci\...
4,1,5,nesciunt quas odio,repudiandae veniam quaerat sunt sed\nalias aut...
